samples/sensor/bq274xx/src/main.c

```diff
@@ -30,95 +30,172 @@ static void do_main(const struct device *dev)
 		full_charge_capacity, remaining_charge_capacity, avg_power,
 		int_temp, current_standby, current_max_load, state_of_health;
 
-	int status = 0;
+	int ret = 0;
 
 	while (1) {
-		status = sensor_sample_fetch(dev);
-		if (status < 0) {
+		ret = sensor_sample_fetch_chan(dev, SENSOR_CHAN_GAUGE_VOLTAGE);
```

Comment:
This is the only change required to actually fix the bug, so other changes should be put into another commit as they are just name changes and error flow changes.